go/distance.go

```diff
@@ -0,0 +1,77 @@
+package cuvs
+
+// #include <stdio.h>
+// #include <stdlib.h>
+// #include <dlpack/dlpack.h>
+// #include <cuda_runtime_api.h>
+// #include <cuvs/core/c_api.h>
+// #include <cuvs/distance/pairwise_distance.h>
+// #include <cuvs/neighbors/brute_force.h>
```

Comment:
For ease of maintenance, can we please create a separate directory for all the testing code?

Reply:
In Go, the convention is to have the tests in the same folder, but with a `_test` suffix, see here: https://blog.jetbrains.com/go/2022/11/22/comprehensive-guide-to-testing-in-go/

The Go toolchain recognizes these files as tests and also excludes them from the compiled binary.